src/Middlewares/Middleware.php

```diff
@@ -93,14 +93,17 @@ function (ResourceContainer $container) {
                 $categoryService = pluginApp(CategoryService::class);
                 $isCategoryPage = $categoryService->getCurrentCategory() !== null && $this->activeOnCatPage;
                 $isInSearchOrCategoryPage = $this->isSearchPage || $isCategoryPage;
+                $currentCategory = $categoryService->getCurrentCategory();
```

Comment:
Please define `$currentCategory` some lines above, so it can be re-used for `$isCategoryPage`